In [3]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [4]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-2/user-2-nonull.csv")
df_origin.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          6658954021          NaN   
1           1                10996          6658965017        Mouse   
2           2                14613          6658968634  Eye Tracker   
3           3                31732          6658985753  Eye Tracker   
4           4                32277          6658986298        Mouse   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-2     Recording3     2025/07/18   
1          JD4  2025/07/18           user-2     Recording3     2025/07/18   
2          JD4  2025/07/18           user-2     Recording3     2025/07/18   
3          JD4  2025/07/18           user-2     Recording3     2025/07/18   
4          JD4  2025/07/18           user-2     Recording3     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN      EyesNotFound   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                       7704.0                     1.0              NaN   
2                       7704.0                     1.0              NaN   
3                       7704.0                     1.0              NaN   
4                       7704.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1            1440.0             897.0  
2               NaN               NaN  
3               NaN               NaN  
4            1439.0             897.0  

[5 rows x 84 columns]

# カラムの整理

In [5]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    16354
Recording timestamp                           16341
Computer timestamp                            16341
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [6]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [7]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-2']
Recording name: ['Recording3']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:13:50.955']
Recording start time UTC: ['07:13:50.955']
Recording duration: [286004]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [15.1]
Average calibration accuracy (degrees): [1.32]
Average calibration accuracy (pixels): [53]
Average validation accuracy (mm): [14]
Average validation precision SD (mm): [17.6]
Average validation precision RMS (mm): [21.8]
Average validation accuracy (degrees): [1.29]
Average validation precision SD (degrees): [1.62]
Average validation precision RMS (degrees): [2.01]
Average validation accuracy (pixels): [49]
Average validation precision

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-2`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:13:50.955`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:13:50.955`|
|Recording duration| 録画時間|`285004`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`15.1`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.32`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`53`pixels|
|Average validation accuracy | 平均バリデーション精度|`14`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.6`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`21.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.29`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.62`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `2.01`degrees|
|Average validation accuracy |平均バリデーション精度| `49`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `76`pixels|

In [8]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          6658954021          NaN   
1           1                10996          6658965017        Mouse   
2           2                14613          6658968634  Eye Tracker   
3           3                31732          6658985753  Eye Tracker   
4           4                32277          6658986298        Mouse   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1                   NaN             NaN         NaN           NaN   
2          6.677293e+09             NaN         NaN           NaN   
3          6.677310e+09             NaN         NaN           NaN   
4                   NaN             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1       EyesNotFound                       7704.0                      1.0   
2       EyesNotFound                       7704.0                      1.0   
3       EyesNotFound                       7704.0                      1.0   
4       EyesNotFound                       7704.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN            1440.0             897.0  
2                         NaN               NaN               NaN  
3                         NaN               NaN               NaN  
4                         NaN            1439.0             897.0  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [9]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             16354
Recording timestamp                    16341
Computer timestamp                     16341
Sensor                                     3
Eyetracker timestamp                   15856
Event                                      9
Event value                                9
Gaze point X                            1769
Gaze point Y                            1035
Gaze point left X                       1726
Gaze point left Y                       1027
Gaze point right X                      1775
Gaze point right Y                      1024
Gaze direction left X                  10834
Gaze direction left Y                  10371
Gaze direction left Z                   5301
Gaze direction right X                 10779
Gaze direction right Y                 10305
Gaze direction right Z                  5398
Pupil diameter left                     1147
Pupil diameter right                     987
Pupil diameter filtered                 1067
Validity l

## Sensor事にデータフレームを分割

In [10]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [11]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0          6658954021    NaN   
281           281              2967295          6661921316    NaN   
5336         5336             86549889          6745503910    NaN   
5343         5343             86630207          6745584228    NaN   
6707         6707            125994380          6784948401    NaN   
6714         6714            126096890          6785050911    NaN   
6733         6733            126513096          6785467117    NaN   
6734         6734            126513096          6785467117    NaN   
7056         7056            131493532          6790447553    NaN   
7057         7057            131493532          6790447553    NaN   
8980         8980            163487731          6822441752    NaN   
8983         8983            163520969          6822474990    NaN   
8984         8984            163520969          6822474990    NaN   
9286         9286            168529652          6827483673    NaN   
9287         9287            168529652          6827483673    NaN   
11195       11195            200264255          6859218276    NaN   
11198       11198            200288917          6859242938    NaN   
11199       11199            200288917          6859242938    NaN   
11501       11501            205296898          6864250919    NaN   
11502       11502            205296898          6864250919    NaN   
13755       13755            242223350          6901177371    NaN   
13757       13757            242247315          6901201336    NaN   
13758       13758            242247315          6901201336    NaN   
14046       14046            247255399          6906209420    NaN   
14047       14047            247255399          6906209420    NaN   
16044       16044            280766529          6939720550    NaN   
16047       16047            280802867          6939756888    NaN   
16048       16048            280802867          6939756888    NaN   
16349       16349            285793525          6944747546    NaN   
16353       16353            286003902          6944957923    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
281                     NaN  Eye tracker Calibration start         NaN   
5336                    NaN                     MouseEvent  Down, Left   
5343                    NaN                     MouseEvent    Up, Left   
6707                    NaN                     MouseEvent  Down, Left   
6714                    NaN                     MouseEvent    Up, Left   
6733                    NaN    Eye tracker Calibration end         NaN   
6734                    NaN             ImageStimulusStart      blank2   
7056                    NaN               ImageStimulusEnd      blank2   
7057                    NaN             ImageStimulusStart  視線誘導問題.001   
8980                    NaN                  KeyboardEvent       Space   
8983                    NaN               ImageStimulusEnd  視線誘導問題.001   
8984                    NaN             ImageStimulusStart      blank2   
9286                    NaN               ImageStimulusEnd      blank2   
9287                    NaN             ImageStimulusStart  視線誘導問題.004   
11195                   NaN                  KeyboardEvent       Space   
11198                   NaN               ImageStimulusEnd  視線誘導問題.004   
11199                   NaN             ImageStimulusStart      blank2   
11501                   NaN               ImageStimulusEnd      blank2   
11502                   NaN             ImageStimulusStart  視線誘導問題.002   
13755                   NaN                  KeyboardEvent       Space   
13757                   NaN               ImageStimulusEnd  視線誘導問題.002   
13758                   NaN             ImageStimulusStart      blank2   
14046                   NaN               ImageStimulusEnd      blank2   
14047            

In [12]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
7057             131493532          6790447553  視線誘導問題.001
8983             163520969          6822474990  視線誘導問題.001
9287             168529652          6827483673  視線誘導問題.004
11198            200288917          6859242938  視線誘導問題.004
11502            205296898          6864250919  視線誘導問題.002
13757            242247315          6901201336  視線誘導問題.002
14047            247255399          6906209420  視線誘導問題.003
16047            280802867          6939756888  視線誘導問題.003


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|7057|131493532|6790447553|視線誘導問題.001|
|8983|163520969|6822474990|視線誘導問題.001|
|9287|168529652|6827483673|視線誘導問題.004|
|11198|200288917|6859242938|視線誘導問題.004|
|11502|205296898|6864250919|視線誘導問題.002|
|13757|242247315|6901201336|視線誘導問題.002|
|14047|247255399|6906209420|視線誘導問題.003|
|16047|280802867|6939756888|視線誘導問題.003|

In [13]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-2/start-stop-timing.csv")

In [15]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
1               1                10996          6658965017  Mouse   
4               4                32277          6658986298  Mouse   
5               5                48138          6659002159  Mouse   
8               8                72142          6659026163  Mouse   
10             10                88195          6659042216  Mouse   
...           ...                  ...                 ...    ...   
11774       11774            209351454          6868305475  Mouse   
11776       11776            209365100          6868319121  Mouse   
11778       11778            209379093          6868333114  Mouse   
11780       11780            209392960          6868346981  Mouse   
11782       11782            209406755          6868360776  Mouse   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1                       NaN   NaN         NaN           NaN           NaN   
4                       NaN   NaN         NaN           NaN           NaN   
5                       NaN   NaN         NaN           NaN           NaN   
8                       NaN   NaN         NaN           NaN           NaN   
10                      NaN   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
11774                   NaN   NaN         NaN           NaN           NaN   
11776                   NaN   NaN         NaN           NaN           NaN   
11778                   NaN   NaN         NaN           NaN           NaN   
11780                   NaN   NaN         NaN           NaN           NaN   
11782                   NaN   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
5                    NaN  ...                    NaN       EyesNotFound   
8                    NaN  ...                    NaN       EyesNotFound   
10                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
11774                NaN  ...                 1080.0            Saccade   
11776                NaN  ...                 1080.0            Saccade   
11778                NaN  ...                 1080.0           Fixation   
11780                NaN  ...                 1080.0           Fixation   
11782                NaN  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                           7704.0                      1.0               NaN   
4                           7704.0                      1.0               NaN   
5                           7704.0                      1.0               NaN   
8                           7704.0                      1.0               NaN   
10                          7704.0                      1.0               NaN   
...                            ...                      ...               ...   
11774                         67.0                    491.0               NaN   
11776                         67.0                    491.0               NaN   
11778                        266.0                    426.0             774.0   
11780                        266.0                    426.0             774.0   
11782                        266.0                    426.0             774.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
4                   NaN                         NaN   
5                   NaN                         NaN   
8                   NaN                         NaN   
10                  NaN                         NaN   
...                 ...                         .

In [14]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                14613          6658968634  Eye Tracker   
3               3                31732          6658985753  Eye Tracker   
6               6                48373          6659002394  Eye Tracker   
7               7                64790          6659018811  Eye Tracker   
9               9                81656          6659035677  Eye Tracker   
...           ...                  ...                 ...          ...   
16347       16347            285771968          6944725989  Eye Tracker   
16348       16348            285788705          6944742726  Eye Tracker   
16350       16350            285805476          6944759497  Eye Tracker   
16351       16351            285821891          6944775912  Eye Tracker   
16352       16352            285838674          6944792695  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              6.677293e+09   NaN         NaN           NaN           NaN   
3              6.677310e+09   NaN         NaN           NaN           NaN   
6              6.677326e+09   NaN         NaN           NaN           NaN   
7              6.677343e+09   NaN         NaN           NaN           NaN   
9              6.677360e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
16347          6.963032e+09   NaN         NaN           NaN           NaN   
16348          6.963049e+09   NaN         NaN           NaN           NaN   
16350          6.963066e+09   NaN         NaN           NaN           NaN   
16351          6.963082e+09   NaN         NaN           NaN           NaN   
16352          6.963099e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       EyesNotFound   
9                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
16347                NaN  ...                 1080.0       EyesNotFound   
16348                NaN  ...                 1080.0       EyesNotFound   
16350                NaN  ...                    NaN       EyesNotFound   
16351                NaN  ...                    NaN       EyesNotFound   
16352                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                           7704.0                      1.0               NaN   
3                           7704.0                      1.0               NaN   
6                           7704.0                      1.0               NaN   
7                           7704.0                      1.0               NaN   
9                           7704.0                      1.0               NaN   
...                            ...                      ...               ...   
16347                         83.0                    191.0               NaN   
16348                         83.0                    191.0               NaN   
16350                         83.0                    191.0               NaN   
16351                         83.0                    191.0               NaN   
16352                         83.0                    191.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
3                   NaN                         NaN   
6                   NaN                         NaN   
7                   NaN                         NaN   
9                   NaN         

## Mouse以外のデータを抽出

In [16]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [17]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [18]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             15885
Recording timestamp                    15876
Computer timestamp                     15876
Sensor                                     2
Eyetracker timestamp                   15856
Event                                      9
Event value                                9
Gaze point X                            1769
Gaze point Y                            1035
Gaze point left X                       1726
Gaze point left Y                       1027
Gaze point right X                      1775
Gaze point right Y                      1024
Gaze direction left X                  10834
Gaze direction left Y                  10371
Gaze direction left Z                   5301
Gaze direction right X                 10779
Gaze direction right Y                 10305
Gaze direction right Z                  5398
Pupil diameter left                     1147
Pupil diameter right                     987
Pupil diameter filtered                 1067
Validity l

In [19]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          6658954021          NaN   
2           2                14613          6658968634  Eye Tracker   
3           3                31732          6658985753  Eye Tracker   
6           6                48373          6659002394  Eye Tracker   
7           7                64790          6659018811  Eye Tracker   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
2          6.677293e+09             NaN         NaN           NaN   
3          6.677310e+09             NaN         NaN           NaN   
6          6.677326e+09             NaN         NaN           NaN   
7          6.677343e+09             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0           NaN                NaN  ...                                  NaN   
2           NaN                NaN  ...                                  NaN   
3           NaN                NaN  ...                                  NaN   
6           NaN                NaN  ...                                  NaN   
7           NaN                NaN  ...                                  NaN   

   Original Media width  Original Media height  Eye movement type  \
0                   NaN                    NaN                NaN   
2                   NaN                    NaN       EyesNotFound   
3                   NaN                    NaN       EyesNotFound   
6                   NaN                    NaN       EyesNotFound   
7                   NaN                    NaN       EyesNotFound   

   Eye movement event duration  Eye movement type index  Fixation point X  \
0                          NaN                      NaN               NaN   
2                       7704.0                      1.0               NaN   
3                       7704.0                      1.0               NaN   
6                       7704.0                      1.0               NaN   
7                       7704.0                      1.0               NaN   

   Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0               NaN                         NaN                         NaN  
2               NaN                         NaN                         NaN  
3               NaN                         NaN                         NaN  
6               NaN                         NaN                         NaN  
7               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [20]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [21]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                14613          6658968634  Eye Tracker   
3               3                31732          6658985753  Eye Tracker   
6               6                48373          6659002394  Eye Tracker   
7               7                64790          6659018811  Eye Tracker   
9               9                81656          6659035677  Eye Tracker   
...           ...                  ...                 ...          ...   
16347       16347            285771968          6944725989  Eye Tracker   
16348       16348            285788705          6944742726  Eye Tracker   
16350       16350            285805476          6944759497  Eye Tracker   
16351       16351            285821891          6944775912  Eye Tracker   
16352       16352            285838674          6944792695  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              6.677293e+09   NaN         NaN           NaN           NaN   
3              6.677310e+09   NaN         NaN           NaN           NaN   
6              6.677326e+09   NaN         NaN           NaN           NaN   
7              6.677343e+09   NaN         NaN           NaN           NaN   
9              6.677360e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
16347          6.963032e+09   NaN         NaN           NaN           NaN   
16348          6.963049e+09   NaN         NaN           NaN           NaN   
16350          6.963066e+09   NaN         NaN           NaN           NaN   
16351          6.963082e+09   NaN         NaN           NaN           NaN   
16352          6.963099e+09   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
3                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       EyesNotFound   
9                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
16347                NaN  ...                 1080.0       EyesNotFound   
16348                NaN  ...                 1080.0       EyesNotFound   
16350                NaN  ...                    NaN       EyesNotFound   
16351                NaN  ...                    NaN       EyesNotFound   
16352                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                           7704.0                      1.0               NaN   
3                           7704.0                      1.0               NaN   
6                           7704.0                      1.0               NaN   
7                           7704.0                      1.0               NaN   
9                           7704.0                      1.0               NaN   
...                            ...                      ...               ...   
16347                         83.0                    191.0               NaN   
16348                         83.0                    191.0               NaN   
16350                         83.0                    191.0               NaN   
16351                         83.0                    191.0               NaN   
16352                         83.0                    191.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
3                   NaN                         NaN   
6                   NaN                         NaN   
7                   NaN                         NaN   
9                   NaN         

## 各問題毎のデータへ切り分け

In [22]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [23]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
11503       11503            205308332          6864262353  Eye Tracker   
11504       11504            205325119          6864279140  Eye Tracker   
11505       11505            205341779          6864295800  Eye Tracker   
11506       11506            205358301          6864312322  Eye Tracker   
11507       11507            205375038          6864329059  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
11503          6.882574e+09   NaN         NaN        1204.0         356.0   
11504          6.882590e+09   NaN         NaN        1201.0         359.0   
11505          6.882607e+09   NaN         NaN        1200.0         352.0   
11506          6.882624e+09   NaN         NaN        1208.0         361.0   
11507          6.882640e+09   NaN         NaN        1204.0         363.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
11503             1205.0  ...                 1080.0           Fixation   
11504             1197.0  ...                 1080.0           Fixation   
11505             1198.0  ...                 1080.0           Fixation   
11506             1209.0  ...                 1080.0           Fixation   
11507             1209.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
11503                        732.0                    412.0            1205.0   
11504                        732.0                    412.0            1205.0   
11505                        732.0                    412.0            1205.0   
11506                        732.0                    412.0            1205.0   
11507                        732.0                    412.0            1205.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
11503             354.0                      0.6274   
11504             354.0                      0.6274   
11505             354.0                      0.6274   
11506             354.0                      0.6274   
11507             354.0                      0.6274   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
11503                      0.3279               NaN               NaN  
11504                      0.3279               NaN               NaN  
11505                      0.3279               NaN               NaN  
11506                      0.3279               NaN               NaN  
11507                      0.3279               NaN               NaN  

[5 rows x 57 columns]

In [24]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             2220
Recording timestamp                    2220
Computer timestamp                     2220
Sensor                                    1
Eyetracker timestamp                   2220
Event                                     1
Event value                               1
Gaze point X                            971
Gaze point Y                            647
Gaze point left X                       995
Gaze point left Y                       684
Gaze point right X                      993
Gaze point right Y                      686
Gaze direction left X                  2055
Gaze direction left Y                  2027
Gaze direction left Z                  1782
Gaze direction right X                 2060
Gaze direction right Y                 2016
Gaze direction right Z                 1803
Pupil diameter left                     465
Pupil diameter right                    368
Pupil diameter filtered                 389
Validity left                   

In [25]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
7058        7058            131505488          6790459509  Eye Tracker   
7059        7059            131521504          6790475525  Eye Tracker   
7060        7060            131538250          6790492271  Eye Tracker   
7061        7061            131555411          6790509432  Eye Tracker   
7062        7062            131571376          6790525397  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
7058          6.808775e+09   NaN         NaN        1168.0         517.0   
7059          6.808791e+09   NaN         NaN        1168.0         515.0   
7060          6.808808e+09   NaN         NaN        1173.0         510.0   
7061          6.808825e+09   NaN         NaN        1172.0         509.0   
7062          6.808841e+09   NaN         NaN        1084.0         487.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
7058             1168.0  ...                 1080.0           Fixation   
7059             1167.0  ...                 1080.0           Fixation   
7060             1168.0  ...                 1080.0           Fixation   
7061             1163.0  ...                 1080.0           Fixation   
7062             1075.0  ...                 1080.0            Saccade   

      Eye movement event duration  Eye movement type index  Fixation point X  \
7058                        216.0                    163.0            1175.0   
7059                        216.0                    163.0            1175.0   
7060                        216.0                    163.0            1175.0   
7061                        216.0                    163.0            1175.0   
7062                         50.0                    215.0               NaN   

      Fixation point Y  Fixation point X (MCSnorm)  \
7058             511.0                      0.6122   
7059             511.0                      0.6122   
7060             511.0                      0.6122   
7061             511.0                      0.6122   
7062               NaN                         NaN   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
7058                      0.4728               NaN               NaN  
7059                      0.4728               NaN               NaN  
7060                      0.4728               NaN               NaN  
7061                      0.4728               NaN               NaN  
7062                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [26]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
14048       14048            247256236          6906210257  Eye Tracker   
14049       14049            247273235          6906227256  Eye Tracker   
14050       14050            247290056          6906244077  Eye Tracker   
14051       14051            247306205          6906260226  Eye Tracker   
14052       14052            247323250          6906277271  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
14048          6.924519e+09   NaN         NaN         961.0         452.0   
14049          6.924536e+09   NaN         NaN         970.0         454.0   
14050          6.924553e+09   NaN         NaN         980.0         448.0   
14051          6.924569e+09   NaN         NaN         970.0         433.0   
14052          6.924586e+09   NaN         NaN         961.0         433.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
14048              948.0  ...                 1080.0           Fixation   
14049              958.0  ...                 1080.0           Fixation   
14050              974.0  ...                 1080.0           Fixation   
14051              966.0  ...                 1080.0           Fixation   
14052              958.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
14048                        266.0                    534.0             965.0   
14049                        266.0                    534.0             965.0   
14050                        266.0                    534.0             965.0   
14051                        266.0                    534.0             965.0   
14052                        266.0                    534.0             965.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
14048             445.0                      0.5024   
14049             445.0                      0.5024   
14050             445.0                      0.5024   
14051             445.0                      0.5024   
14052             445.0                      0.5024   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
14048                      0.4119               NaN               NaN  
14049                      0.4119               NaN               NaN  
14050                      0.4119               NaN               NaN  
14051                      0.4119               NaN               NaN  
14052                      0.4119               NaN               NaN  

[5 rows x 57 columns]

In [27]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
9288        9288            168531520          6827485541  Eye Tracker   
9289        9289            168548418          6827502439  Eye Tracker   
9290        9290            168564806          6827518827  Eye Tracker   
9291        9291            168581341          6827535362  Eye Tracker   
9292        9292            168598341          6827552362  Eye Tracker   

      Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
9288          6.845799e+09   NaN         NaN         726.0         312.0   
9289          6.845816e+09   NaN         NaN         729.0         312.0   
9290          6.845832e+09   NaN         NaN         732.0         316.0   
9291          6.845849e+09   NaN         NaN         731.0         318.0   
9292          6.845866e+09   NaN         NaN         730.0         311.0   

      Gaze point left X  ...  Original Media height  Eye movement type  \
9288              748.0  ...                 1080.0           Fixation   
9289              745.0  ...                 1080.0           Fixation   
9290              759.0  ...                 1080.0           Fixation   
9291              744.0  ...                 1080.0           Fixation   
9292              749.0  ...                 1080.0           Fixation   

      Eye movement event duration  Eye movement type index  Fixation point X  \
9288                        499.0                    290.0             731.0   
9289                        499.0                    290.0             731.0   
9290                        499.0                    290.0             731.0   
9291                        499.0                    290.0             731.0   
9292                        499.0                    290.0             731.0   

      Fixation point Y  Fixation point X (MCSnorm)  \
9288             310.0                      0.3809   
9289             310.0                      0.3809   
9290             310.0                      0.3809   
9291             310.0                      0.3809   
9292             310.0                      0.3809   

      Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
9288                      0.2874               NaN               NaN  
9289                      0.2874               NaN               NaN  
9290                      0.2874               NaN               NaN  
9291                      0.2874               NaN               NaN  
9292                      0.2874               NaN               NaN  

[5 rows x 57 columns]

In [28]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.004
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.004']
  Presented Media name: ['視線誘導問題.004.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presen

In [29]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns
視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-2`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:13:50.955`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:13:50.955`|
|Recording duration| 録画時間|`285004`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`15.1`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.32`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`53`pixels|
|Average validation accuracy | 平均バリデーション精度|`14`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.6`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`21.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.29`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.62`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `2.01`degrees|
|Average validation accuracy |平均バリデーション精度| `49`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `76`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [31]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
7058        7058            131505488          6790459509   
7059        7059            131521504          6790475525   
7060        7060            131538250          6790492271   
7061        7061            131555411          6790509432   
7062        7062            131571376          6790525397   
...          ...                  ...                 ...   
8977        8977            163439155          6822393176   
8978        8978            163456260          6822410281   
8979        8979            163472207          6822426228   
8981        8981            163489073          6822443094   
8982        8982            163506183          6822460204   

      Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
7058          6.808775e+09        1168.0         517.0             1168.0   
7059          6.808791e+09        1168.0         515.0             1167.0   
7060          6.808808e+09        1173.0         510.0             1168.0   
7061          6.808825e+09        1172.0         509.0             1163.0   
7062          6.808841e+09        1084.0         487.0             1075.0   
...                    ...           ...           ...                ...   
8977          6.840707e+09        1490.0         506.0             1482.0   
8978          6.840724e+09        1489.0         507.0             1488.0   
8979          6.840740e+09        1492.0         506.0             1492.0   
8981          6.840757e+09        1495.0         505.0             1491.0   
8982          6.840774e+09        1482.0         513.0             1476.0   

      Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
7058              509.0              1168.0               524.0  ...   
7059              510.0              1170.0               520.0  ...   
7060              507.0              1178.0               512.0  ...   
7061              509.0              1181.0               510.0  ...   
7062              481.0              1094.0               493.0  ...   
...                 ...                 ...                 ...  ...   
8977              487.0              1497.0               524.0  ...   
8978              482.0              1491.0               531.0  ...   
8979              477.0              1492.0               535.0  ...   
8981              475.0              1499.0               535.0  ...   
8982              490.0              1488.0               535.0  ...   

      Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
7058                       0.4713                        0.6084   
7059                       0.4725                        0.6093   
7060                       0.4696                        0.6136   
7061                       0.4709                        0.6150   
7062                       0.4458                        0.5696   
...                           ...                           ...   
8977                       0.4511                        0.7795   
8978                       0.4466                        0.7766   
8979                       0.4415                        0.7772   
8981                       0.4397                        0.7805   
8982                       0.4537                        0.7748   

      Gaze point right Y (MCSnorm)  Eye movement type  \
7058                        0.4853           Fixation   
7059                        0.4818           Fixation   
7060                        0.4740           Fixation   
7061                        0.4718           Fixation   
7062                        0.4569            Saccade   
...                            ...                ...   
8977                        0.4856           Fixation   
8978                        0.4914           Fixation   
8979                        0.4958           Fixation   
8981                        0.4953           Fixation   
8982                        0.4954           Fixati

In [32]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-2/user-2-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-2`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:13:50.955`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:13:50.955`|
|Recording duration| 録画時間|`285004`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`15.1`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.32`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`53`pixels|
|Average validation accuracy | 平均バリデーション精度|`14`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.6`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`21.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.29`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.62`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `2.01`degrees|
|Average validation accuracy |平均バリデーション精度| `49`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `76`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [33]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
9288         9288            168531520          6827485541   
9289         9289            168548418          6827502439   
9290         9290            168564806          6827518827   
9291         9291            168581341          6827535362   
9292         9292            168598341          6827552362   
...           ...                  ...                 ...   
11192       11192            200216917          6859170938   
11193       11193            200233237          6859187258   
11194       11194            200249621          6859203642   
11196       11196            200266860          6859220881   
11197       11197            200283156          6859237177   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
9288           6.845799e+09         726.0         312.0              748.0   
9289           6.845816e+09         729.0         312.0              745.0   
9290           6.845832e+09         732.0         316.0              759.0   
9291           6.845849e+09         731.0         318.0              744.0   
9292           6.845866e+09         730.0         311.0              749.0   
...                     ...           ...           ...                ...   
11192          6.877482e+09         840.0         797.0              864.0   
11193          6.877499e+09         838.0         815.0              864.0   
11194          6.877515e+09         837.0         790.0              873.0   
11196          6.877532e+09         932.0         768.0              954.0   
11197          6.877549e+09        1169.0         655.0             1159.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
9288               319.0               704.0               306.0  ...   
9289               322.0               714.0               302.0  ...   
9290               321.0               706.0               312.0  ...   
9291               324.0               719.0               313.0  ...   
9292               317.0               710.0               305.0  ...   
...                  ...                 ...                 ...  ...   
11192              823.0               816.0               771.0  ...   
11193              851.0               813.0               778.0  ...   
11194              803.0               800.0               778.0  ...   
11196              789.0               910.0               747.0  ...   
11197              682.0              1178.0               627.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
9288                        0.2953                        0.3665   
9289                        0.2984                        0.3718   
9290                        0.2969                        0.3678   
9291                        0.2995                        0.3747   
9292                        0.2931                        0.3697   
...                            ...                           ...   
11192                       0.7618                        0.4251   
11193                       0.7884                        0.4235   
11194                       0.7432                        0.4167   
11196                       0.7310                        0.4739   
11197                       0.6319                        0.6138   

       Gaze point right Y (MCSnorm)  Eye movement type  \
9288                         0.2829           Fixation   
9289                         0.2795           Fixation   
9290                         0.2889           Fixation   
9291                         0.2899           Fixation   
9292                         0.2822           Fixation   
...                             ...                ...   
11192                        0.7140       Unclassified   
11193                        0.7200       Unclassified   
11194                        0.7206       Unclassified   
11196                        0.6919            Sacc

In [34]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-2/user-2-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-2`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:13:50.955`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:13:50.955`|
|Recording duration| 録画時間|`285004`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`15.1`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.32`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`53`pixels|
|Average validation accuracy | 平均バリデーション精度|`14`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.6`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`21.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.29`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.62`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `2.01`degrees|
|Average validation accuracy |平均バリデーション精度| `49`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `76`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [35]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
11503       11503            205308332          6864262353   
11504       11504            205325119          6864279140   
11505       11505            205341779          6864295800   
11506       11506            205358301          6864312322   
11507       11507            205375038          6864329059   
...           ...                  ...                 ...   
13751       13751            242168411          6901122432   
13752       13752            242185038          6901139059   
13753       13753            242201458          6901155479   
13754       13754            242218329          6901172350   
13756       13756            242234961          6901188982   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
11503          6.882574e+09        1204.0         356.0             1205.0   
11504          6.882590e+09        1201.0         359.0             1197.0   
11505          6.882607e+09        1200.0         352.0             1198.0   
11506          6.882624e+09        1208.0         361.0             1209.0   
11507          6.882640e+09        1204.0         363.0             1209.0   
...                     ...           ...           ...                ...   
13751          6.919431e+09        1500.0         166.0             1517.0   
13752          6.919448e+09        1436.0         177.0             1448.0   
13753          6.919464e+09        1430.0         175.0             1443.0   
13754          6.919481e+09        1424.0         177.0             1437.0   
13756          6.919498e+09        1422.0         176.0             1432.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
11503              338.0              1203.0               374.0  ...   
11504              340.0              1205.0               378.0  ...   
11505              331.0              1203.0               372.0  ...   
11506              341.0              1206.0               381.0  ...   
11507              333.0              1198.0               393.0  ...   
...                  ...                 ...                 ...  ...   
13751              132.0              1483.0               200.0  ...   
13752              148.0              1424.0               206.0  ...   
13753              153.0              1416.0               197.0  ...   
13754              160.0              1411.0               193.0  ...   
13756              158.0              1413.0               195.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
11503                       0.3130                        0.6265   
11504                       0.3146                        0.6274   
11505                       0.3068                        0.6264   
11506                       0.3156                        0.6282   
11507                       0.3085                        0.6242   
...                            ...                           ...   
13751                       0.1221                        0.7726   
13752                       0.1372                        0.7415   
13753                       0.1421                        0.7377   
13754                       0.1486                        0.7348   
13756                       0.1458                        0.7358   

       Gaze point right Y (MCSnorm)  Eye movement type  \
11503                        0.3460           Fixation   
11504                        0.3498           Fixation   
11505                        0.3448           Fixation   
11506                        0.3524           Fixation   
11507                        0.3640           Fixation   
...                             ...                ...   
13751                        0.1856            Saccade   
13752                        0.1904            Saccade   
13753                        0.1826           Fixation   
13754                        0.1791           Fixat

In [36]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-2/user-2-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-2`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:13:50.955`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:13:50.955`|
|Recording duration| 録画時間|`285004`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`15.1`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.32`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`53`pixels|
|Average validation accuracy | 平均バリデーション精度|`14`mm|
|Average validation precision SD | 平均バリデーション予測SD|`17.6`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`21.6`mm|
|Average validation accuracy |平均バリデーション精度| `1.29`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `1.62`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `2.01`degrees|
|Average validation accuracy |平均バリデーション精度| `49`pixels|
|Average validation precision SD |平均バリデーション予測SD| `61`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `76`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [37]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
14048       14048            247256236          6906210257   
14049       14049            247273235          6906227256   
14050       14050            247290056          6906244077   
14051       14051            247306205          6906260226   
14052       14052            247323250          6906277271   
...           ...                  ...                 ...   
16041       16041            280729735          6939683756   
16042       16042            280746464          6939700485   
16043       16043            280763320          6939717341   
16045       16045            280779658          6939733679   
16046       16046            280796336          6939750357   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
14048          6.924519e+09         961.0         452.0              948.0   
14049          6.924536e+09         970.0         454.0              958.0   
14050          6.924553e+09         980.0         448.0              974.0   
14051          6.924569e+09         970.0         433.0              966.0   
14052          6.924586e+09         961.0         433.0              958.0   
...                     ...           ...           ...                ...   
16041          6.957990e+09        1120.0         229.0             1116.0   
16042          6.958007e+09        1110.0         228.0             1110.0   
16043          6.958024e+09        1118.0         228.0             1116.0   
16045          6.958040e+09        1116.0         233.0             1113.0   
16046          6.958057e+09        1104.0         228.0             1092.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
14048              467.0               973.0               437.0  ...   
14049              469.0               981.0               439.0  ...   
14050              470.0               986.0               426.0  ...   
14051              458.0               974.0               409.0  ...   
14052              467.0               964.0               398.0  ...   
...                  ...                 ...                 ...  ...   
16041              237.0              1124.0               222.0  ...   
16042              248.0              1110.0               209.0  ...   
16043              234.0              1120.0               222.0  ...   
16045              233.0              1119.0               233.0  ...   
16046              216.0              1115.0               239.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
14048                       0.4323                        0.5067   
14049                       0.4341                        0.5111   
14050                       0.4349                        0.5135   
14051                       0.4242                        0.5075   
14052                       0.4329                        0.5023   
...                            ...                           ...   
16041                       0.2193                        0.5852   
16042                       0.2297                        0.5780   
16043                       0.2165                        0.5832   
16045                       0.2153                        0.5831   
16046                       0.2003                        0.5808   

       Gaze point right Y (MCSnorm)  Eye movement type  \
14048                        0.4044           Fixation   
14049                        0.4066           Fixation   
14050                        0.3942           Fixation   
14051                        0.3783           Fixation   
14052                        0.3686           Fixation   
...                             ...                ...   
16041                        0.2053            Saccade   
16042                        0.1934           Fixation   
16043                        0.2059           Fixation   
16045                        0.2161           Fixat

In [38]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-2/user-2-pic-3.csv")